pkg/kubelet/cm/node_container_manager_linux.go

```diff
@@ -60,6 +60,37 @@ func (cm *containerManagerImpl) createNodeAllocatableCgroups() error {
 	return nil
 }
 
+func (cm *containerManagerImpl) updateInternalCapacity(rl v1.ResourceList) bool {
+	changed := false
+
+	for k, v := range rl {
+		if _, ok := cm.internalCapacity[k]; !ok {
+			continue
+		}
+
+		if !v.Equal(cm.internalCapacity[k]) {
+			klog.V(5).Infof("node allocatable resource %q has changed", k)
+			cm.internalCapacity[k] = v
+			changed = true
+		}
+	}
+
+	return changed
+}
+
+// UpdateNodeAllocatableCgroups updates cgroups for node allocatable resources.
+func (cm *containerManagerImpl) UpdateNodeAllocatableCgroups(rl v1.ResourceList) error {
+
+	if cm.updateInternalCapacity(rl) {
```

Comment:
this would have some false positives for extended integer resources or device plugins, but its not a major concern.

Reply:
so calling this in the main kubelet status loop could block up to 1m if the update fails.  see line 137 where we sleep on failure.

Reply:
But isn't this sleep run in an async goroutine?

Reply:
@derekwaynecarr the time.Sleep is in a goroutine... That being said... The time.Sleep is extremely odd within that go routine.